{{cookiecutter.project_slug}}/src/{{cookiecutter.project_module}}/errorhandlers.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2018, Novo Nordisk Foundation Center for Biosustainability,
+# Technical University of Denmark.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Error handlers and custom exceptions.
+
+http://flask.pocoo.org/docs/1.0/errorhandling/
+http://flask.pocoo.org/docs/1.0/patterns/apierrors/
+"""
+
+import logging
+import sys
+
+from flask import jsonify
+
+from werkzeug.exceptions import HTTPException
+from werkzeug.routing import RoutingException
+
+
+logger = logging.getLogger(__name__)
+
+
+def init_app(app):
+    @app.errorhandler(422)
+    def handle_webargs_error(error):
+        """
+        Handle webargs parser errors.
+
+        Ensures a JSON response including all error messages produced from the
+        parser, that we know to be available on the error object.
+
+        Note that this sets some constrains for manually throwing 422 exceptions
```

Comment:
I think we're good here since we should never raise 422 ourselves. 422 is from the WebDAV extension https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422 and I think such cases should simply be 400.

Reply:
I am admittedly a bit curious as to why webargs choose to use 422.

Reply:
It was the default response chosen by Ruby on Rails when request parameters could not be parsed. At least, that's where I learned to use it and then had to unlearn it :smile: